src/Orchard.Web/Modules/Orchard.MediaLibrary/ViewModels/MediaManagerFolderEditViewModel.cs

```diff
@@ -1,11 +1,21 @@
-﻿using System.Collections.Generic;
+﻿using System;
+using System.Collections.Generic;
 using System.ComponentModel.DataAnnotations;
-using Orchard.MediaLibrary.Models;
+using System.Web.Configuration;
 
 namespace Orchard.MediaLibrary.ViewModels {
     public class MediaManagerFolderEditViewModel {
         [Required]
         public string Name { get; set; }
         public string FolderPath { get; set; }
+        public string InvalidCharactersPattern {
+            get {
+                string invalidChars = ((SystemWebSectionGroup)WebConfigurationManager.OpenWebConfiguration(null).GetSectionGroup("system.web")).HttpRuntime.RequestPathInvalidCharacters;
```

Comment:
This code is replicated 3 times.

I'd suggest that you just create a static Regex that checks for any known chars, without having to read the web.config. It would be perfect but it adds unnecessary complexity in this case.